src/oscar/apps/dashboard/catalogue/views.py

```diff
@@ -163,7 +163,7 @@ def apply_search(self, queryset):
                     Q(id__in=matches_upc.values('id')) | Q(id__in=matches_upc.values('parent_id')))
 
         if data.get('title'):
-            queryset = queryset.filter(title__icontains=data['title'])
+            queryset = queryset.filter(Q(title__icontains=data['title']) | Q(children__title__icontains=data['title']))
 
         return queryset
```

Comment:
I think we need to return `queryset.distinct()` here since it is now possible to return duplicates (e.g., if a child contains the same title as its parent, or if they share part of an SKU).